license.md

```diff
@@ -60,9 +60,9 @@ You need not contribute prototype changes, extensions, or applications that you
 
 ## Applications
 
-The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software, and to prohibit use of this software to create competing substitutes for this software that are not contributed.
+The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software, minimizing duplication of effort across competing substitutes.
 
-You need not contribute software that only invokes this software's functionality through the interfaces this software exposes, without exposing this software's interfaces and functionality to other software.
+You need not contribute any application that only invokes this software's functionality through the interfaces this software exposes, without exposing this software's interfaces or functionality to users or other software to such an extent that it becomes a practical substitute for this software.
```

Comment:
I think I liked "software" better than "any application", in the first part of the sentence&mdash;application could be interpreted to mean an "app" as opposed to a higher-level service that is backed by a lower-level service provided by this software.

I like "users or other software", since I feel like we do want to cover apps.

"Practical substitute" seems kinda vague, but maybe in an intentional way? Like putting "reasonable" all over the place in contracts, that's the part you _want_ to be hashing out in court/at the negotiating table. (At least, based on my impression of lawyering.) I do wonder if we could be _slightly_ less vague, maybe something like, "generally understood in the industry to be a substitute" or "generally accepted by the industry as a substitute" or "commonly used and recommended in the industry as a potential substitute", or something.

I kind of like that last one. Still not a lawyer but my understanding of what we're aiming for is to, in e.g. the case of the bioinformatics database and general database, we're aiming to calibrate the wording so that the negotiating leverage we'd have is appropriate for how arguable it is that the bioinformatics database is a substitute vs an application of the general database. (_Appropriate_ leverage, because if it's too little then this license is toothless, and if it's too much then people don't want to build their application with our software even though we would want them to.) So, if the bioinformatics database can only store RNA & DNA strings and one eclectic hacker builds their email marketing app with a custom ORM (Object-RNA/DNA Mapping), then it'd still be hard to argue it's "commonly used and recommended" as a substitute. Whereas if the bioinformatics database merely optimized RNA & DNA string search, then you'd see bioinformatics researchers who were previously using the general database migrate to the bioinformatics-specific one, even if the API were slightly incompatible.

Reply:
> I think I liked "software" better than "any application", in the first part of the sentence

You're right. And I've made this mistake before, in Parity.

My intent here wasn't to evoke the general meaning of "application", but rather to pick a name for "the things that don't have to be contributed back".

> I like "users or other software", since I feel like we do want to cover apps.

Exactly what I had in mind.

> "Practical substitute" seems kinda vague, but maybe in an intentional way?

Correct, alas.

I'm convinced there's no objectively verifiable, doubtlessly administrable fact of development, like whether a file was excerpted or modified under MPLv2, that we can rely on to implement the idea of an application, as distinct from a change, extension, or wrapper. So we're going to have to accept some fuzziness around the rule.

> we're aiming to calibrate the wording so that the negotiating leverage we'd have is appropriate for how arguable it is that the bioinformatics database is a substitute vs an application of the general database

Correct. And very lawyerly of you to say so.

The general reflex when accepting intentional fuzziness is to set a kind of "meta rule" for interpretation. If a case is close, which way do we err?

I'm thinking more and more that I should simply accept the fuzziness of competition/substitution, and set that kind of meta rule. Probably to err on the side of requiring contribution.

Reply:
I've added a meta rule favoring contribution in 1f8ff7d7a6a7b751ea2b65b5291aa5c58cb65a06